src/o365/teams/commands/deeplink/tab-generate.ts

```diff
@@ -0,0 +1,188 @@
+import commands from '../../commands';
+import GlobalOptions from '../../../../GlobalOptions';
+import {
+  CommandOption, CommandValidate
+} from '../../../../Command';
+import GraphCommand from '../../../base/GraphCommand';
+import Utils from '../../../../Utils';
+import request from '../../../../request';
+import { Tab } from '../../Tab';
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+export enum TabTypeOptions {
+  Static = "Static",
+  Configurable = "Configurable"
+}
+
+export interface Deeplink {
+  deeplink: string;
+}
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  teamId: string;
+  channelId: string;
+  tabId: string;
+  label: string;
+  tabType: string;
+}
+
+class TeamsDeeplinkTabGenerateCommand extends GraphCommand {
+  public get name(): string {
+    return `${commands.TEAMS_DEEPLINK_TAB_GENERATE}`;
+  }
+
+  public get description(): string {
+    return 'Generates a Microsoft Teams deep link from an existing Tab in a Channel';
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: () => void): void {
+    const requestOptions: any = {
+      url: `${this.resource}/v1.0/teams/${encodeURIComponent(args.options.teamId)}/channels/${encodeURIComponent(args.options.channelId)}/tabs/${encodeURIComponent(args.options.tabId)}?$expand=teamsApp`,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      json: true
+    }
+
+    request
+      .get<Tab>(requestOptions)
+      .then((res: Tab): void => {
+        let appId: string = res.teamsApp.id;
+        let entityId: string = res.configuration.entityId;
+        let contentUrl: string = encodeURIComponent(res.webUrl);
+        let deeplink: Deeplink = { deeplink: "" };
+
+        let tabTypeInput: string = args.options.tabType ? args.options.tabType.trim() : TabTypeOptions.Static;
+
+        if (TabTypeOptions[(tabTypeInput as keyof typeof TabTypeOptions)].valueOf() == TabTypeOptions.Configurable) {         
+            let context: string = `{"channelId": "${args.options.channelId}"}`;
+            deeplink = { deeplink: `https://teams.microsoft.com/l/entity/${appId}/${entityId}?webUrl=${contentUrl}&label=${args.options.label}&context=${context}` };
+            cmd.log(deeplink);
+        }
+        else if (TabTypeOptions[(tabTypeInput as keyof typeof TabTypeOptions)].valueOf() == TabTypeOptions.Static) {
+            deeplink = { deeplink: `https://teams.microsoft.com/l/entity/${appId}/${entityId}?webUrl=${contentUrl}&label=${args.options.label}` };
+            cmd.log(deeplink);
```

Comment:
As `args.options.label` property could include special characters or spaces, and will be used in a URL, we should encode the value.